### PR TITLE
feat!: upgrade to pydantic v2

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: tokenlists/typing.py [hunk 1]
```diff
@@ -180,18 +184,24 @@ class Config:
...
             raise ValueError(f"'{v}' is not a valid URI")
 
         return v
 
-    def dict(self, *args, **kwargs) -> dict:
-        data = super().dict(*args, **kwargs)
-        # NOTE: This was the easiest way to make sure this property returns isoformat
-        data["timestamp"] = self.timestamp.isoformat()
+    def dict(self, *args, **kwargs):
+        return self.model_dump(*args, **kwargs)
```
Comment (fubuloubu) [inline tokenlists/typing.py]: Do we want to remove this method and make this a breaking change PR? (I mean it probably should be a breaking change PR to mitigate any down-stream effects of v2-only dependency)

**Review by fubuloubu — APPROVED** [review]
Any notes on performance?

**Comment by fubuloubu** [comment]
v1 fuzzing CI takes ~6.5 minutes: https://github.com/ApeWorX/py-tokenlists/actions/runs/5696191075/job/15440825648

v2 shaves 45 seconds off that time

File: tokenlists/typing.py (hunk 1 above)
Comment (antazoey) [reply tokenlists/typing.py]: Realizing that we have to remove this as it causes problems, removed here: [9683fe9](https://github.com/ApeWorX/py-tokenlists/pull/36/commits/9683fe91767dff1112196b9605f24b598fcd7434)

Comment (antazoey) [reply tokenlists/typing.py]: And ya we can make these breaking change PRs

**Pull request merged**